server/src/com/cloud/api/ApiServer.java

```diff
@@ -264,10 +266,11 @@ public void handleAsyncJobPublishEvent(String subject, String senderAddress, Obj
         String info = job.getCmdInfo();
         String cmdEventType = "unknown";
         if (info != null) {
-            String marker = "\"cmdEventType\"";
-            int begin = info.indexOf(marker);
-            if (begin >= 0) {
-                cmdEventType = info.substring(begin + marker.length() + 2, info.indexOf(",", begin) - 1);
+            Type type = new TypeToken<Map<String, Object>>(){}.getType();
+            Map<String, Object> cmdInfo = ApiGsonHelper.getBuilder().create().fromJson(info, type);
```

Comment:
@alexandrelimassantana You're right, it was unnecessary. I've updated the pull request accordingly.